### DISCUSSION TITLE
请问简繁转换如何使用 我下了最新的文件夹现在连字体都改不了了

### DISCUSSION THREAD

**Discussion by 2650951697** [Tony15246/uosc_danmaku#92] [Q&A]
添加uosc_danmaku.conf直接失效了是怎么回事啊

**Comment by Tony15246** [comment]
可以提供以下更详细的信息吗？

你下载的是releasev1.2.0的代码，还是主分支最新的源代码？你使用的是windows还是linux操作系统？

> 添加uosc_danmaku.conf直接失效了是怎么回事啊

你说的`失效`的具体表现是什么？是指你在`uosc_danmaku.conf`中进行任何设置都无法生效吗？

**Reply by Tony15246** [reply]
你可以在讨论区上传一下你的mpv的运行日志文件；和你配置的`uosc_danmaku.conf`文件，方便我帮你找找看出了什么问题。

**Reply by 2650951697** [reply]
两个我都下了 就只有releasev1.2.0的能用 用的win11 失效的表现就是那个 无法生效 话说我纯纯编程小白完全不知道怎么看那个mpv的运行日志文件啊

**Reply by 2650951697** [reply]
![QQ20241214-113007](https://github.com/user-attachments/assets/58a95fd8-dc1f-4cc7-8ff6-544043f9277f)

**Reply by Tony15246** [reply]
> 两个我都下了 就只有releasev1.2.0的能用 用的win11 失效的表现就是那个 无法生效 话说我纯纯编程小白完全不知道怎么看那个mpv的运行日志文件啊

1. 只有releasev1.2.0能用，主分支最新的源代码无法正常运行吗？主分支最新源代码无法使用的具体表现是什么，哪个功能出现故障？

简繁转换的功能在releasev1.2.0发布的时候还没有，只有主分支最新的源代码才有。如果想使用简繁转换功能的话，那就要先解决运行主分支最新的源代码时的问题。

2. 所以你说的`无法生效`，是指`chConvert`选项和`fontname`选项，这两个选项设置了也不会生效，其余功能和选项都能正常运行是吗？

首先，关于`fontname`选项，脚本只是负责指定字体，具体在系统中寻找字体和处理字体的行为是由mpv封装的libass管理的，并且和mpv字体相关设置以及系统字体设置有关。字体设置的行为不归插件管理，我这里只能给你一些排查的建议。

先检查fontname是否配置成了你想要设置的字体的family的名称。如果确认配置无误，检查`mpv.conf`中是否设置了`osd-font-provider`选项，如果设置了，将该选项删除后再查看字体设置是否能够生效。

其次，关于`chConvert`选项不能生效的问题，就如我在1中说的，目前只有主分支最新的源代码才有简繁转换功能

3. 关于如何查看mpv的运行日志文件的问题。这个很简单，不需要懂编程。你查看你的`mpv.conf`文件，检查你是否配置了`log-file`选项。如果配置了，那么mpv在每次运行的时候，都会把该次运行的日志记录到`log-file`选项中配置的路径，只要去对应的路径找到该日志文件就行。如果没有配置过这个选项，那么在`mpv.conf`文件中添加下面这一行

```
log-file=~~/mpv.log
```

这样mpv在每次运行的时候，都会把该次运行的日志记录到mpv配置文件夹的根目录底下，文件名就叫`mpv.log`

## 总结

解决`chConvert`选项无法生效的问题，首先需要使用主分支最新的源代码，然后运行mpv播放视频并使用本插件以模拟故障发生的情形，最后寻找mpv记录的日志文件并且发给我，我来排查出现的问题。

`fontname`选项无法生效则需要你自己排查mpv的字体相关设置和windows的系统字体设置，我能给到的建议已经在2中都说过了。

**Reply by 2650951697** [reply]
[mpv.log](https://github.com/user-attachments/files/18135703/mpv.log)

**Reply by 2650951697** [reply]
字体那个我用release1.2就有用啊 用源代码没用 那应该是源代码的问题吧？

**Reply by Tony15246** [reply]
我看了一眼你的运行日志。你把插件在的文件夹命名为了`uosc_danmaku_main`，mpv还以为插件的名字叫`uosc_danmaku_main`，跑去找名字叫`uosc_danmaku_main.conf`的配置文件了。那难怪你在`uosc_danmaku.conf`里的设置没法生效了😂，mpv根本不会去读这个文件。

那字体设置不生效也是文件夹命名错误，导致的配置文件不生效引起的附带的问题，重命名一下应该就都解决了

**Reply by Tony15246** [reply]
把插件所在文件夹重命名成`uosc_danmaku`就好了

**Reply by 2650951697** [reply]
这样啊 原来如此 犯了这么低级的错误啊 主要是我下来就是这个文件 我下意识以为单纯移过去不用改设置就好了 之前还把你添加的那个简繁转换的更新代码操作用1.2的版本复制了那个操作结果弹幕打都打不开了 看来还得是你们专业的来 牵一发而动全身啊

**Reply by 2650951697** [reply]
感谢感谢 我这个确实有点搞笑了

**Reply by Tony15246** [reply]
不用谢ξ( ✿＞◡❛)

谁都有犯低级错误的时候，都是人之常情😆

**Reply by 2650951697** [reply]
大佬好暖心 也谢谢大佬做的这个插件解决了我外挂弹幕的需求(ﾉ´▽｀)ﾉ♪

**Comment by Tony15246** [comment]
> 请问简繁转换如何使用

就在`uosc_danmaku.conf`里设置`chConvert`选项就行

这样设置就是全转换为简体

```
chConvert=1
```

这样设置就是全转换为繁体

```
chConvert=2
```

**Reply by 2650951697** [reply]
这个我看见了 就是配置了这个也都生效不了 连带着字体也生效不了

**Reply by 2650951697** [reply]
你好 我改了文件名之后弹幕字体是正常了 那个vf滤镜也生效了 但是还是没有转简体是怎么回事

**Reply by 2650951697** [reply]
配置的是1

**Reply by 2650951697** [reply]
[mpv.log](https://github.com/user-attachments/files/18135937/mpv.log)

**Reply by Tony15246** [reply]
> 你好 我改了文件名之后弹幕字体是正常了 那个vf滤镜也生效了 但是还是没有转简体是怎么回事

我看一下日志哈

**Reply by Tony15246** [reply]
奇怪，报错说缺文件了。

你截个图我看一下你的`F:/下载/mpv_config-2024.12.04/portable_config/scripts/uosc_danmaku/bin/OpenCC_Windows`文件夹底下文件全不全

**Reply by 2650951697** [reply]
是全的吧 我重下了一个对比了下
![8e07ce1c-ad29-4022-8d0c-b5a7ff9b59c0](https://github.com/user-attachments/assets/8ce9a76c-78c2-4a0a-bd84-316390c0d018)

**Reply by Tony15246** [reply]
是全的，没问题。我大概猜测到是什么问题了。我需要你配合我做个测试，验证一下我的假设。假如这个bug确实如和我想的一样的话，我待会儿修复一下这个bug。

一会儿我发给你一个新的`render.lua`文件，你用这个文件去替换掉原本的`F:/下载/mpv_config-2024.12.04/portable_config/scripts/uosc_danmaku/render.lua`文件。然后你再尝试一下简繁转换功能是否正常。

**Reply by 2650951697** [reply]
OK 谢谢大佬

**Reply by Tony15246** [reply]
[render.zip](https://github.com/user-attachments/files/18136157/render.zip)
github不支持传.lua后缀的文件，我把文件压缩到了zip里，打开压缩包就是用来替换的`render.lua`文件

你替换一下试试看，如果还有问题就把新的mpv.log发给我

**Reply by 2650951697** [reply]
ok 我试试

**Reply by 2650951697** [reply]
[mpv.log](https://github.com/user-attachments/files/18136170/mpv.log)

**Reply by 2650951697** [reply]
还是不行诶

**Reply by Tony15246** [reply]
奇怪了，我原本怀疑是不是相对路径的问题。但是给你的新代码指定了绝对路径竟然还是找不到文件。

我怀疑是不是opencc对中文路径的支持有问题。你试试看把F盘那个`下载`文件夹重命名成英文试试看呢？比如说改成`Downloads`

**Reply by 2650951697** [reply]
牛逼！解决了 就是刚刚打开文件要重新选择打开方式 要在电脑里重新选那个文件勾选 但是解决了 话说重新勾选的原因是不是就是你指定了绝对路径所以打不开啊？应该不是吧 你那个只是换了脚本的文件

**Reply by Tony15246** [reply]
>  就是刚刚打开文件要重新选择打开方式 

因为你mpv也放到原先的`下载`文件夹下了，所以你把这个文件夹重命名了以后，windows不确定你的mpv放哪了，所以让你重新选了一下打开方式，相当于让你重新手动指定了一下mpv的位置

**Reply by 2650951697** [reply]
噢噢 原来是这样 怪不得两个mpv的图标都消失了 好了 谢谢大佬的答疑解惑 辛苦了 搞了半天都是我自己的原因 和你这个程序没关系啊 不过应该也是让你知道了这个opencc对中文路径的支持有问题 后面你也可以把这个注意事项加进去让后来的人知道最好不要不要把mpv放在中文文件夹哈哈 我是因为个人使用习惯 习惯用简介的中文创建文件夹 比如番剧我就放在叫番的文件夹里哈哈 没想到还有会遇到不支持中文文件夹的时候 这也是给我撞上小概率事件了感觉

**Reply by Tony15246** [reply]
确实，我准备更新一下readme加一个对这个问题的提醒。这又是一个经典的程序员酒吧炒饭问题，我在自己的linux和同学的windows电脑上都测试过了，没发现什么问题。结果还是有这种小概率事件导致的问题😂

**Reply by 2650951697** [reply]
哈哈 刚刚搜了一下你说的这个酒吧炒饭 是说程序员进了酒吧点了个炒饭还是写的程序酒吧里有人点了个炒饭然后就崩溃了啊？ 我看有人是这样说的 百度搜的也不尽相同 https://user.guancha.cn/main/child-comments?id=25833305 到底具体是指什么意思啊？

**Reply by Tony15246** [reply]
> 是说程序员进了酒吧点了个炒饭还是写的程序酒吧里有人点了个炒饭然后就崩溃了啊？

就是这个笑话，原文是这样的，你应该也搜到了

> 
    一个测试工程师走进一家酒吧，要了一杯啤酒；
    一个测试工程师走进一家酒吧，要了一杯咖啡；
    一个测试工程师走进一家酒吧，要了0.7杯啤酒；
    一个测试工程师走进一家酒吧，要了-1杯啤酒；
    一个测试工程师走进一家酒吧，要了232杯啤酒；
    一个测试工程师走进一家酒吧，要了一杯洗脚水；
    一个测试工程师走进一家酒吧，要了一杯蜥蜴；
    一个测试工程师走进一家酒吧，要了一份asdfQwer@24dg!&*(@；
    一个测试工程师走进一家酒吧，什么也没要；
    一个测试工程师走进一家酒吧，又走出去又从窗户进来又从后门出去从下水道钻进来；
    一个测试工程师走进一家酒吧，又走出去又进来又出去又进来又出去，最后在外面把老板打了一顿；
    一个测试工程师走进一家酒吧，要了一杯烫烫烫的锟斤拷；
    一个测试工程师走进一家酒吧，要了NaN杯Null；
    一个测试工程师冲进一家酒吧，要了500T啤酒咖啡洗脚水野猫狼牙棒奶茶；
    一个测试工程师把酒吧拆了；
    一个测试工程师化装成老板走进一家酒吧，要了500杯啤酒并且不付钱；
    一万个测试工程师在酒吧门外呼啸而过；
     一个测试工程师走进一家酒吧，"< script >alert("要了一杯酒");< /script >"
    一个测试工程师走进一家酒吧，要了一杯啤酒';DROP TABLE 酒吧；
    测试工程师们满意地离开了酒吧。
    然后一名顾客点了一份炒饭，酒吧炸了。

这是一个很经典的程序员笑话。表达的意思就是你永远不知道用户会以什么样的方式使用你做的软件。就算你自以为自己算无遗策，预料到了并且处理了所有稀奇古怪的刁钻使用场景；用户总会用出各种你意想不到的问题，这些问题可能都是很简单的小问题，可是在用户把软件用出这些问题之前，站在程序员的视角上是想破脑袋也想不到还有这些奇葩问题的。

这个笑话在作为笑话的同时，对软件测试这个领域的从业人员也是有学习和教育意义的。

**Reply by 2650951697** [reply]
原来如此 你这个版本搜的比我更全 配合你的解释我明白了 虽然我不是程序员 但是确实能感到这个笑话很具备启示意义啊

**Comment by 2650951697** [comment]
ok 我等会试试 字体那个我用release1.2就有用啊 用源代码没用 那应该是源代码的问题吧？




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月14日(星期六) 下午2:48
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Tony15246/uosc_danmaku] 请问简繁转换如何使用 我下了最新的文件夹现在连字体都改不了了 (Discussion #92)














  
两个我都下了 就只有releasev1.2.0的能用 用的win11 失效的表现就是那个 无法生效 话说我纯纯编程小白完全不知道怎么看那个mpv的运行日志文件啊
   
只有releasev1.2.0能用，主分支最新的源代码无法正常运行吗？主分支最新源代码无法使用的具体表现是什么，哪个功能出现故障？
  
简繁转换的功能在releasev1.2.0发布的时候还没有，只有主分支最新的源代码才有。如果想使用简繁转换功能的话，那就要先解决运行主分支最新的源代码时的问题。
  
所以你说的无法生效，是指chConvert选项和fontname选项，这两个选项设置了也不会生效，其余功能和选项都能正常运行是吗？
  
首先，关于fontname选项，脚本只是负责指定字体，具体在系统中寻找字体和处理字体的行为是由mpv封装的libass管理的，并且和mpv字体相关设置以及系统字体设置有关。字体设置的行为不归插件管理，我这里只能给你一些排查的建议。
 
先检查fontname是否配置成了你想要设置的字体的family的名称。如果确认配置无误，检查mpv.conf中是否设置了osd-font-provider选项，如果设置了，将该配置删除后再查看字体设置是否能够生效。
 
其次，关于chConvert选项不能生效的问题，就如我在1中说的，目前只有主分支最新的源代码才有简繁转换功能
  
关于如何查看mpv的运行日志文件的问题。这个很简单，不需要懂编程。你查看你的mpv.conf文件，检查你是否配置了log-file选项。如果配置了，那么mpv在每次运行的时候，都会把该次运行的日志记录到log-file选项中配置的路径，只要去对应的路径找到该日志文件就行。如果没有配置过这个选项，那么在mpv.conf文件中添加下面这一行
  log-file=~~/mpv.log  
这样mpv在每次运行的时候，都会把该次运行的日志记录到mpv配置文件夹的根目录底下，文件名就叫mpv.log
 
总结
 
解决chConvert选项无法生效的问题，首先需要使用主分支最新的源代码，然后运行mpv播放视频并使用本插件以模拟故障发生的情形，最后寻找mpv记录的日志文件并且发给我，我来排查出现的问题。
 
fontname选项无法生效则需要你自己排查mpv的字体相关设置和windows的系统字体设置，我能给到的建议已经在2中都说过了。
